### PR TITLE
Bluetooth: Controller: Enable ISO support in the SDC

### PR DESCRIPTION
**Pull request by rugeGerritsen**
This commit enables the experimental ISO support in the SoftDevice Controller. See the changelog for more details.

### CODE REVIEW

File: subsys/bluetooth/controller/hci_driver.c [hunk 1]
```diff
@@ -163,6 +163,60 @@ BUILD_ASSERT(!IS_ENABLED(CONFIG_BT_PERIPHERAL) ||
...
+#define SDC_CIS_COUNT CONFIG_BT_CTLR_CONN_ISO_STREAMS
+#else
+#define SDC_MEM_CIG   0
+#define SDC_MEM_CIS   0
+#define SDC_CIS_COUNT 0
+#endif
+
+#if defined(CONFIG_BT_CTLR_SYNC_ISO)
+#define SDC_MEM_BIS_SINK \
+	SDC_MEM_PER_BIG(CONFIG_BT_CTLR_SYNC_ISO_STREAM_COUNT) +					\
```
Comment (sean-madigan) [inline subsys/bluetooth/controller/hci_driver.c:180]: I'm currently trying to merge a fix to change this to CONFIG_BT_CTLR_SCAN_SYNC_ISO_SET

Comment (sean-madigan) [reply subsys/bluetooth/controller/hci_driver.c:180]: and here https://github.com/nrfconnect/sdk-nrf/pull/12508/files#diff-ca9ee596a131877b2cbc629ccf2e965015db73c3d5cc74be46a064c4bfbf81a8R1019

File: subsys/bluetooth/controller/hci_internal.c [hunk 2]
```diff
@@ -1234,6 +1301,151 @@ static uint8_t le_controller_cmd_put(uint8_t const * const cmd,
...
+			(sdc_hci_cmd_le_setup_iso_data_path_t const *)cmd_params,
+			(sdc_hci_cmd_le_setup_iso_data_path_return_t *)event_out_params);
+
+	case SDC_HCI_OPCODE_CMD_LE_REMOVE_ISO_DATA_PATH:
+		*param_length_out += sizeof(sdc_hci_cmd_le_remove_iso_data_path_return_t);
+		return sdc_hci_cmd_le_remove_iso_data_path(
+			(sdc_hci_cmd_le_remove_iso_data_path_t const *)cmd_params,
+			(sdc_hci_cmd_le_remove_iso_data_path_return_t *)event_out_params);
+
+	case SDC_HCI_OPCODE_CMD_LE_ISO_TEST_END:
```
Comment (Tschet1) [inline subsys/bluetooth/controller/hci_internal.c:1430]: we can do that in a separate PR, but should we consider ifdef-ing out the test commands?

File: subsys/bluetooth/controller/hci_internal.c [hunk 3]
```diff
@@ -1234,6 +1301,151 @@ static uint8_t le_controller_cmd_put(uint8_t const * const cmd,
...
+		*param_length_out += sizeof(sdc_hci_cmd_le_iso_test_end_return_t);
+		return sdc_hci_cmd_le_iso_test_end(
+			(sdc_hci_cmd_le_iso_test_end_t const *)cmd_params,
+			(sdc_hci_cmd_le_iso_test_end_return_t *)event_out_params);
+
+	case SDC_HCI_OPCODE_CMD_LE_SET_HOST_FEATURE:
```
Comment (Tschet1) [inline subsys/bluetooth/controller/hci_internal.c:1436]: Should this be in a CONFIG_BT_CTLR_SET_HOST_FEATURE ?

File: subsys/bluetooth/controller/hci_internal.c [hunk 4]
```diff
@@ -40,6 +40,7 @@ static bool command_generates_command_complete_event(uint16_t hci_opcode)
 {
 	switch (hci_opcode) {
 	case SDC_HCI_OPCODE_CMD_LC_DISCONNECT:
+	case SDC_HCI_OPCODE_CMD_CB_WRITE_CONN_ACCEPT_TIMEOUT:
```
Comment (Tschet1) [inline subsys/bluetooth/controller/hci_internal.c:43]: The SDC_HCI_OPCODE_CMD_CB_WRITE_CONN_ACCEPT_TIMEOUT creates a command complete

File: subsys/bluetooth/controller/hci_internal.c (hunk 2 above)
Comment (rugeGerritsen) [reply subsys/bluetooth/controller/hci_internal.c:1430]: I agree. We could consider adding more ifdefs for both the ISO test commands and also the Create CIG/BIG Test commands, Read Link quality and others

**Review by Tschet1 — CHANGES_REQUESTED** [review]
.

File: subsys/bluetooth/controller/hci_internal.c (hunk 3 above)
Comment (Tschet1) [reply subsys/bluetooth/controller/hci_internal.c:1436]: should it also be outside the CONFIG_BT_CTLR_ISO, to be aligned with https://github.com/nrfconnect/sdk-nrf/pull/12508/files#diff-f5bf9d2614b19c06ca472417c8004daeae8125af4a59c8ec6983180c9ddae3daR560 ?

**Review by Tschet1 — CHANGES_REQUESTED** [review]

**Review by sean-madigan — APPROVED** [review]

**Review by Tschet1 — APPROVED** [review]

**Review by rtreccani — APPROVED** [review]
LGTM
